### PR TITLE
Fix test failures caused by connection resets (#1167).

### PR DESCRIPTION
**Pull request by cesarbs**
#1167

@halter73

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs [hunk 1]
```diff
@@ -592,7 +593,7 @@ public void ConnectionCanReadAndWrite(TestServiceContext testContext)
                         "GET / HTTP/1.1",
                         "Connection: close",
                         "",
-                        "a");
+                        "");
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs]: RecieveForcedEnd

Comment (halter73) [inline test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs]: ReceiveForcedEnd

**Comment by cesarbs** [comment]
Updated.

File: test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs [hunk 2]
```diff
@@ -608,7 +609,7 @@ public void ConnectionCanReadAndWrite(TestServiceContext testContext)
                         "GET / HTTP/1.0",
                         "",
                         "a");
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs]: I assume we don't want to send the extra "a" here too.

**Comment by cesarbs** [comment]
Updated & rebased.

**Comment by benaadams** [comment]
This mean we can have Loopback Fast Path back? ;-)

**Comment by cesarbs** [comment]
@benaadams Unrelated, to the best of my knowledge.

**Comment by halter73** [comment]
:shipit:

**Pull request merged**